src/script/treeView/lib/wrapper.ts

```diff
@@ -111,30 +111,65 @@ function updateTreeViewAndSearchResults(searchStr: string) {
     }
     searchStrOldValue = searchStr;
 }
-
-export function toggleSelectedNode(jsonPath) {
-    const focusNode = document.getElementById(jsonPath);
-    if (focusNode == null) {
-        return "";
+function focusGivenNode(focusNode) {
+    if (focusNode.className !== "node_selected") {
+        document.getElementsByClassName("node_selected")[0].className = "node";
+        focusNode.className = "node_selected";
     }
-    const selectedNode = document.getElementsByClassName("node_selected");
-    if (selectedNode.length > 0) {
-        selectedNode[0].className = "node";
+}
+function expandUpwardTree(jsonPath) {
+    const jsonPathList = jsonPath.split("/");
+    const partialPath = [""];
+    for (let i = 1; i < jsonPathList.length - 1; i++) {
+        partialPath[partialPath.length] = jsonPathList[i];
+        const jsonTreePathString = partialPath.join("/");
+        const ulElement = document.getElementById(jsonTreePathString);
+        const img = ulElement.firstElementChild;
+        const spanElement = img.nextSibling;
+        if (img.id !== "toggle_off") {
+            const evt = new Event("dblclick");
+            spanElement.dispatchEvent(evt);
+        }
     }
-    const nodeToSelect = document.getElementById(jsonPath).parentNode;
-    (nodeToSelect.childNodes[1] as HTMLSpanElement).className = "node_selected";
-    let nodeToExpand = document.getElementById(jsonPath);
-    while (nodeToExpand.id !== escape(rootNode.path)) {
-        nodeToExpand = (nodeToExpand.parentNode.parentNode) as HTMLUListElement;
-        (nodeToExpand as HTMLUListElement).style.display = "block";
-        const nodeExpColImg = nodeToExpand.getElementsByTagName("img")[0];
-        nodeExpColImg.id = "toggle_off";
-        nodeExpColImg.src = "images/collapse.png";
+    const selectNodeParent = document.getElementById(jsonPath);
+    const selectedNode = selectNodeParent.getElementsByTagName("span")[0];
+    selectedNode.scrollIntoView({ behavior: "smooth", block: "nearest", inline: "start" });
+    focusGivenNode(selectedNode);
+}
+
+export function toggleSelectedNode(jsonPath) {
+    const decodedJsonPath = decodeURIComponent(jsonPath);
+    if (decodedJsonPath.split("/").pop() === "control") {
+        document.getElementsByClassName("node_selected")[0].className = "node";
+    } else {
+        const focusNode = document.getElementById(decodedJsonPath);
+        let focusNodeTemp = focusNode;
+        /* If the tree strcuture for the file that we request is not expanded then,
+        we expand it first. Then, we highlight(node_selected) the file in the file-pane.
+        Else, we just do the node selection in the file pane for the requested file,
+        as the file tree is already open */
+        if (focusNodeTemp == null) {
+            expandUpwardTree(decodedJsonPath);
```

Comment:
You could consider renaming `expandUpwardTree()` to `createSubTreeAndExpandBranchForPath()`. And you can extract the code in the else block into `expandBranchForPath()`.